### PR TITLE
7.post api articles article id comments

### PR DESCRIPTION
**Pull request by dave-hickman**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -173,35 +172,69 @@ describe("GET /api/articles/:article_id/comments", () => {
...
     return request(app)
-    .get("/api/articles/999/comments")
-    .expect(404)
-    .then((response) => {
-      expect(response.body).toEqual({err: "Resource not found"})
+      .post("/api/articles/1/comments")
+      .send(newComment)
+      .expect(201)
+      .then((response) => {
+        expect(typeof response.body.comment[0].author).toBe("string");
```
Comment (Augs0) [inline __tests__/app.test.js]: We would want to check for the actual values as we know them from the newComment variable

File: __tests__/app.test.js [hunk 2]
```diff
@@ -173,35 +172,69 @@ describe("GET /api/articles/:article_id/comments", () => {
...
-      expect(response.body.comments.length).toBe(0)
-    })
-    
+      .post("/api/articles/dogs/comments")
+      .send(newComment)
+      .expect(400)
+      .then((response) => {
+        expect(response.body).toEqual({ msg: "Invalid Input!" });
+      });
   });
```
Comment (Augs0) [inline __tests__/app.test.js:239]: This isa great start but there are a number of other tests to consider here:

> what if I send through additional properties on my comment. We should ignore these
> what if I'm missing a property e.g. I don't send a body
> what if I use a username that doesn't exist?

File: models/articles_model.js [hunk 3]
```diff
@@ -9,21 +10,28 @@ exports.selectArticle = (articleID) => {
       [articleID]
     )
     .then((article) => {
-      if (article.rows.length === 0){
-        return Promise.reject({status: 404, msg: `No user found for user_id: ${articleID}!`})
+      if (article.rows.length === 0) {
+        return Promise.reject({
+          status: 404,
+          msg: `No user found for user_id: ${articleID}!`,
```
Comment (Augs0) [inline models/articles_model.js]: This doesn't seem to quite match up. You're talking about users and articles. Which one are we checking?

File: playground.txt [hunk 4]
```diff
@@ -3,8 +3,8 @@
 > psql -f ./db/playground.sql
 
 You are now connected to database "nc_news_test" as user "apple".
- comment_id |                                                                                         body                                                                                         | article_id |    author     | votes |     created_at
```
Comment (Augs0) [inline playground.txt:6]: We don't need to include this file in commits

**Review by Augs0 — CHANGES_REQUESTED** [review]
Hey Dave, a great attempt so far. Just a few more tests to add to get this ticket up to where it needs to be, but a fab job so far!

File: __tests__/app.test.js [hunk 5]
```diff
@@ -237,4 +237,35 @@ describe("POST /api/articles/:article_id/comments", () => {
...
+  });
+  it('should ignore additional properties in the comment', () => {
+    const newComment = { username: "icellusedkars", body: "cba to register", votes: 1};
+    return request(app)
+      .post("/api/articles/1/comments")
+      .send(newComment)
+      .expect(400)
+      .then((response) => {
+        expect(response.body).toEqual({ err: "Invalid inputs provided!" });
+      });
```
Comment (robert112233) [inline __tests__/app.test.js:264]: When August mentioned ignoring additional keys, we would still consider this a happy path. The extra keys will be ignored and a 201 status will be provided as normal. The user has done the _right_ thing, just with some additional _wrong_ things which can simply be ignored, so no error 👍

File: models/articles_model.js [hunk 6]
```diff
@@ -44,7 +44,18 @@ exports.selectComments = (articleID) => {
...
+    return Promise.reject({
+      status:400, 
+      msg: 'Missing inputs!'
+    })
+  }
+  if (Object.keys(commentInfo).length > 2){
+    return Promise.reject({
+      status:400, 
+      msg: 'Invalid inputs provided!'
+    })
```
Comment (robert112233) [inline models/articles_model.js]: We can remove this (see above comment) 😃

**Review by robert112233 — CHANGES_REQUESTED** [review]
Amend the faulty test and submit another PR for review. Great work on Augusts' feedback!

**Review by robert112233 — APPROVED** [review]
Great work , merge!

**Pull request merged**